src/Core/Scanner.vala

```diff
@@ -44,7 +47,7 @@ namespace PantheonCalculator.Core {
             Evaluation e = new Evaluation ();
 
             for (int i = 0; input.get_next_char (ref index, out c); i++) {
-                if (c != ' ') {
+                if (c != ' ' && c != scanner.separator_symbol) {
```

Comment:
Should this really be accessed from the instance like this?

Reply:
Are you referring to scanner.separator_symbol? Why is that an issue?

Reply:
Well because scanner is an object instance whereas the method is static. I wouldn't count this as orthodox, even though it is possible.